Repository: plone.browserlayer


Branch: refs/heads/master
Date: 2022-11-08T11:43:56+01:00
Author: Maurits van Rees (mauritsvanrees) <maurits@vanrees.org>
Commit: https://github.com/plone/plone.browserlayer/commit/fd93bbd2b3be6da2e80155da54abf1397d94a483

Fix 'KeyError: file' in browser tests on Python 3.11.

Fixes part of https://github.com/plone/Products.CMFPlone/issues/3663

See https://github.com/plone/Products.CMFPlone/pull/3673 for a similar fix I did in CMFPlone.
In that case I chose to rewrite the doctest as a unit test.
The main fix in both cases is setting `browser.raiseHttpErrors = False`.
This means that you get the original `zException.NotFound` instead of an HTTPError.
The HTTPError is the one that gives problems in Python 3.11 because too many things have changed in exception handling, mostly for improved display of the traceback.

Files changed:
A news/3663.bugfix
M plone/browserlayer/README.rst
M setup.py

b'diff --git a/news/3663.bugfix b/news/3663.bugfix\nnew file mode 100644\nindex 0000000..c32b9f0\n--- /dev/null\n+++ b/news/3663.bugfix\n@@ -0,0 +1,2 @@\n+Fix \'KeyError: file\' in browser tests on Python 3.11.\n+[maurits]\ndiff --git a/plone/browserlayer/README.rst b/plone/browserlayer/README.rst\nindex 5962037..df8a429 100644\n--- a/plone/browserlayer/README.rst\n+++ b/plone/browserlayer/README.rst\n@@ -11,14 +11,18 @@ Before the product is installed, we cannot view this:\n     >>> IMyProductLayer in utils.registered_layers()\n     False\n \n-    >>> from plone.testing import z2\n-\n-    >>> from plone.testing.z2 import Browser\n+    >>> try:\n+    ...     from plone.testing.zope import Browser\n+    ... except ImportError:\n+    ...     from plone.testing.z2 import Browser\n     >>> browser = Browser(layer[\'app\'])\n+    >>> browser.handleErrors = False\n+    >>> browser.raiseHttpErrors = False\n+    >>> browser.open(layer[\'portal\'].absolute_url())\n     >>> browser.open(layer[\'portal\'].absolute_url() + \'/@@layer-test-view\')\n     Traceback (most recent call last):\n     ...\n-    urllib.error.HTTPError: HTTP Error 404: Not Found\n+    zExceptions.NotFound: http://nohost/plone/@@layer-test-view\n \n We can view a view registered for the default layer, though:\n \n@@ -61,7 +65,7 @@ It is also possible to uninstall a layer:\n     >>> browser.open(layer[\'portal\'].absolute_url() + \'/@@layer-test-view\')\n     Traceback (most recent call last):\n     ...\n-    urllib.error.HTTPError: HTTP Error 404: Not Found\n+    zExceptions.NotFound: http://nohost/plone/@@layer-test-view\n \n GenericSetup support\n --------------------\n@@ -148,7 +152,7 @@ as expected:\n     >>> browser.open(layer[\'portal\'].absolute_url() + \'/@@layer-test-view\')\n     Traceback (most recent call last):\n     ...\n-    urllib.error.HTTPError: HTTP Error 404: Not Found\n+    zExceptions.NotFound: http://nohost/plone/@@layer-test-view\n \n     >>> browser.open(layer[\'portal\'].absolute_url() + \'/@@standard-test-view\')\n     >>> print(browser.contents)\ndiff --git a/setup.py b/setup.py\nindex 0266892..7cc487a 100644\n--- a/setup.py\n+++ b/setup.py\n@@ -20,6 +20,7 @@\n         "Framework :: Plone :: 5.0",\n         "Framework :: Plone :: 5.1",\n         "Framework :: Plone :: 5.2",\n+        "Framework :: Plone :: 6.0",\n         "Framework :: Plone :: Core",\n         "Framework :: Zope2",\n         "Framework :: Zope :: 4",\n@@ -30,6 +31,9 @@\n         "Programming Language :: Python :: 3.6",\n         "Programming Language :: Python :: 3.7",\n         "Programming Language :: Python :: 3.8",\n+        "Programming Language :: Python :: 3.9",\n+        "Programming Language :: Python :: 3.10",\n+        "Programming Language :: Python :: 3.11",\n     ],\n     keywords=\'plone browser layer\',\n     author=\'Plone Foundation\',\n'

Repository: plone.browserlayer


Branch: refs/heads/master
Date: 2022-11-08T11:59:15+01:00
Author: Maurits van Rees (mauritsvanrees) <maurits@vanrees.org>
Commit: https://github.com/plone/plone.browserlayer/commit/58ed6e58c2a15d38d02dd431b4d75e2de6f270d0

Drop support for Plone 5 and Python 2.

Files changed:
A news/600.breaking
M plone/browserlayer/README.rst
M setup.py

b'diff --git a/news/600.breaking b/news/600.breaking\nnew file mode 100644\nindex 0000000..e30e3ad\n--- /dev/null\n+++ b/news/600.breaking\n@@ -0,0 +1 @@\n+Drop support for Plone 5 and Python 2.  [maurits]\ndiff --git a/plone/browserlayer/README.rst b/plone/browserlayer/README.rst\nindex df8a429..4423a5e 100644\n--- a/plone/browserlayer/README.rst\n+++ b/plone/browserlayer/README.rst\n@@ -11,10 +11,7 @@ Before the product is installed, we cannot view this:\n     >>> IMyProductLayer in utils.registered_layers()\n     False\n \n-    >>> try:\n-    ...     from plone.testing.zope import Browser\n-    ... except ImportError:\n-    ...     from plone.testing.z2 import Browser\n+    >>> from plone.testing.zope import Browser\n     >>> browser = Browser(layer[\'app\'])\n     >>> browser.handleErrors = False\n     >>> browser.raiseHttpErrors = False\ndiff --git a/setup.py b/setup.py\nindex 7cc487a..b5efe8a 100644\n--- a/setup.py\n+++ b/setup.py\n@@ -2,12 +2,12 @@\n from setuptools import find_packages\n from setuptools import setup\n \n-version = \'2.2.5.dev0\'\n+version = \'3.0.0.dev0\'\n \n setup(\n     name=\'plone.browserlayer\',\n     version=version,\n-    description="Browser layer management for Zope 2 applications",\n+    description="Browser layer management for Zope applications",\n     long_description=(\n         open("README.rst").read() +\n         "\\n" +\n@@ -17,19 +17,12 @@\n         "Development Status :: 5 - Production/Stable",\n         "Environment :: Web Environment",\n         "Framework :: Plone",\n-        "Framework :: Plone :: 5.0",\n-        "Framework :: Plone :: 5.1",\n-        "Framework :: Plone :: 5.2",\n         "Framework :: Plone :: 6.0",\n         "Framework :: Plone :: Core",\n-        "Framework :: Zope2",\n         "Framework :: Zope :: 4",\n         "License :: OSI Approved :: GNU General Public License v2 (GPLv2)",\n         "Operating System :: OS Independent",\n         "Programming Language :: Python",\n-        "Programming Language :: Python :: 2.7",\n-        "Programming Language :: Python :: 3.6",\n-        "Programming Language :: Python :: 3.7",\n         "Programming Language :: Python :: 3.8",\n         "Programming Language :: Python :: 3.9",\n         "Programming Language :: Python :: 3.10",\n@@ -38,7 +31,7 @@\n     keywords=\'plone browser layer\',\n     author=\'Plone Foundation\',\n     author_email=\'plone-developers@lists.sourceforge.net\',\n-    url=\'https://pypi.org/project/plone.browserlayer\',\n+    url=\'https://github.com/plone/plone.browserlayer\',\n     license=\'GPL version 2\',\n     packages=find_packages(),\n     namespace_packages=[\'plone\'],\n@@ -56,6 +49,6 @@\n         \'zope.traversing >= 3.9.0\',\n         \'Products.CMFCore\',\n         \'Products.GenericSetup>=1.4\',\n-        \'Zope2\',\n+        \'Zope\',\n     ],\n )\n'

Repository: plone.browserlayer


Branch: refs/heads/master
Date: 2022-11-08T16:54:43+01:00
Author: Maurits van Rees (mauritsvanrees) <maurits@vanrees.org>
Commit: https://github.com/plone/plone.browserlayer/commit/f378d72274cb324e669485ad991ccb08f358ad32

Merge pull request #10 from plone/maurits-python311-traceback-error-in-tests

Fix 'KeyError: file' in browser tests on Python 3.11. Drop 5.2.

Files changed:
A news/3663.bugfix
A news/600.breaking
M plone/browserlayer/README.rst
M setup.py

b'diff --git a/news/3663.bugfix b/news/3663.bugfix\nnew file mode 100644\nindex 0000000..c32b9f0\n--- /dev/null\n+++ b/news/3663.bugfix\n@@ -0,0 +1,2 @@\n+Fix \'KeyError: file\' in browser tests on Python 3.11.\n+[maurits]\ndiff --git a/news/600.breaking b/news/600.breaking\nnew file mode 100644\nindex 0000000..e30e3ad\n--- /dev/null\n+++ b/news/600.breaking\n@@ -0,0 +1 @@\n+Drop support for Plone 5 and Python 2.  [maurits]\ndiff --git a/plone/browserlayer/README.rst b/plone/browserlayer/README.rst\nindex 5962037..4423a5e 100644\n--- a/plone/browserlayer/README.rst\n+++ b/plone/browserlayer/README.rst\n@@ -11,14 +11,15 @@ Before the product is installed, we cannot view this:\n     >>> IMyProductLayer in utils.registered_layers()\n     False\n \n-    >>> from plone.testing import z2\n-\n-    >>> from plone.testing.z2 import Browser\n+    >>> from plone.testing.zope import Browser\n     >>> browser = Browser(layer[\'app\'])\n+    >>> browser.handleErrors = False\n+    >>> browser.raiseHttpErrors = False\n+    >>> browser.open(layer[\'portal\'].absolute_url())\n     >>> browser.open(layer[\'portal\'].absolute_url() + \'/@@layer-test-view\')\n     Traceback (most recent call last):\n     ...\n-    urllib.error.HTTPError: HTTP Error 404: Not Found\n+    zExceptions.NotFound: http://nohost/plone/@@layer-test-view\n \n We can view a view registered for the default layer, though:\n \n@@ -61,7 +62,7 @@ It is also possible to uninstall a layer:\n     >>> browser.open(layer[\'portal\'].absolute_url() + \'/@@layer-test-view\')\n     Traceback (most recent call last):\n     ...\n-    urllib.error.HTTPError: HTTP Error 404: Not Found\n+    zExceptions.NotFound: http://nohost/plone/@@layer-test-view\n \n GenericSetup support\n --------------------\n@@ -148,7 +149,7 @@ as expected:\n     >>> browser.open(layer[\'portal\'].absolute_url() + \'/@@layer-test-view\')\n     Traceback (most recent call last):\n     ...\n-    urllib.error.HTTPError: HTTP Error 404: Not Found\n+    zExceptions.NotFound: http://nohost/plone/@@layer-test-view\n \n     >>> browser.open(layer[\'portal\'].absolute_url() + \'/@@standard-test-view\')\n     >>> print(browser.contents)\ndiff --git a/setup.py b/setup.py\nindex 0266892..b5efe8a 100644\n--- a/setup.py\n+++ b/setup.py\n@@ -2,12 +2,12 @@\n from setuptools import find_packages\n from setuptools import setup\n \n-version = \'2.2.5.dev0\'\n+version = \'3.0.0.dev0\'\n \n setup(\n     name=\'plone.browserlayer\',\n     version=version,\n-    description="Browser layer management for Zope 2 applications",\n+    description="Browser layer management for Zope applications",\n     long_description=(\n         open("README.rst").read() +\n         "\\n" +\n@@ -17,24 +17,21 @@\n         "Development Status :: 5 - Production/Stable",\n         "Environment :: Web Environment",\n         "Framework :: Plone",\n-        "Framework :: Plone :: 5.0",\n-        "Framework :: Plone :: 5.1",\n-        "Framework :: Plone :: 5.2",\n+        "Framework :: Plone :: 6.0",\n         "Framework :: Plone :: Core",\n-        "Framework :: Zope2",\n         "Framework :: Zope :: 4",\n         "License :: OSI Approved :: GNU General Public License v2 (GPLv2)",\n         "Operating System :: OS Independent",\n         "Programming Language :: Python",\n-        "Programming Language :: Python :: 2.7",\n-        "Programming Language :: Python :: 3.6",\n-        "Programming Language :: Python :: 3.7",\n         "Programming Language :: Python :: 3.8",\n+        "Programming Language :: Python :: 3.9",\n+        "Programming Language :: Python :: 3.10",\n+        "Programming Language :: Python :: 3.11",\n     ],\n     keywords=\'plone browser layer\',\n     author=\'Plone Foundation\',\n     author_email=\'plone-developers@lists.sourceforge.net\',\n-    url=\'https://pypi.org/project/plone.browserlayer\',\n+    url=\'https://github.com/plone/plone.browserlayer\',\n     license=\'GPL version 2\',\n     packages=find_packages(),\n     namespace_packages=[\'plone\'],\n@@ -52,6 +49,6 @@\n         \'zope.traversing >= 3.9.0\',\n         \'Products.CMFCore\',\n         \'Products.GenericSetup>=1.4\',\n-        \'Zope2\',\n+        \'Zope\',\n     ],\n )\n'

